### DISCUSSION TITLE
World Building

### DISCUSSION THREAD

**Discussion by EdwardProj** [smlu/ProjectMarduk#8] [General]
Hello,

I'm a huge fan of this game and have long waited to pick through the files. In the past I've been able to extract the files, play sounds and view textures. I've also edited cog files but not had a great deal of luck with much else. Now even better tools have been created and I can edit textures and the 3D objects, that the game will recognize.

I'm curious to know what the next steps will be, especially in terms of world building. Do you think it will be possible to build levels 3D without having to hand code everything in a text editor? Would also be interested in more edited files, would love to play around adding more enemies, placing the jeep where it shouldn't be and learn how the rail cart works.

Thanks for your great work!

**Comment by smlu** [comment]
A level editor would be great but I'm afraid this project is far from it atm.
Fyi, if you are comfortable with blender scripts, I made script which leverages [blender-sith](https://github.com/smlu/blender-sith) addon and can export 3D object(s) to `NDY` level format. The script can be found here:  [https://github.com/Jones3D-The-Infernal-Engine/Blender-scripts/blob/main/obj_to_ndy.py](https://github.com/Jones3D-The-Infernal-Engine/Blender-scripts/blob/main/obj_to_ndy.py).
Note that the script is incomplete and if you have more than 1 level sector you'll have to manually add the adjoins. You'll also have to manually add the template and thing sections. The docs for `NDY` file format can be found here: https://github.com/Jones3D-The-Infernal-Engine/Documentation/blob/main/ndy.md

**Comment by Aetherium1947** [comment]
Hi Edward.

We also have a GitHub for creations: https://github.com/Jones3D-The-Infernal-Engine
Exemple of levels: https://github.com/Jones3D-The-Infernal-Engine/test-levels
Exemple of models: https://github.com/Jones3D-The-Infernal-Engine/3D-models-HD-recreation

If you have Discord, a server about Jones3D exist: https://discord.com/invite/W7EV2nhKgh

**Reply by EdwardProj** [reply]
Thank you for both of your contributions.
I've done quite a bit of playing around with editing the levels, but so far I have not managed to make the jeep work or even attempted the raft or the rail cart yet.
So far I have been loading the game and typing Coords in the console, but that is quite a process, especially in areas I cannot get to, is there an easier way to do this?
I'll definitely take a look through the discord server though.

**Reply by Aetherium1947** [reply]
Oh sorry, I was busy with the end of the year and completely forgot to reply to you.

There is no other way to make world geometry (for now). We only have the script for Blender. When you understand how a level is made, it's easy to place one or several functionnal jeep, raft or minecar in any level. I could tell you how to do that if you really want.
